diff --git a/arch/arm/plat-aspeed/Makefile b/arch/arm/plat-aspeed/Makefile
index 55d9a0d..6b077f6 100644
--- a/arch/arm/plat-aspeed/Makefile
+++ b/arch/arm/plat-aspeed/Makefile
@@ -32,6 +32,11 @@ ifeq ($(CONFIG_CMM),y)
   PLATFB = 1
 endif
 
+ifeq ($(CONFIG_VOYAGER),y)
+  obj-y += dev-uart.o dev-i2c-setup-mux.o dev-i2c-cmm.o
+  PLATFB = 1
+endif
+
 ifeq ($(CONFIG_WEDGE), y)
   obj-y += dev-uart.o
   PLATFB = 1
diff --git a/arch/arm/plat-aspeed/dev-eth.c b/arch/arm/plat-aspeed/dev-eth.c
index 7e28d00..d86e515 100644
--- a/arch/arm/plat-aspeed/dev-eth.c
+++ b/arch/arm/plat-aspeed/dev-eth.c
@@ -41,7 +41,7 @@
 #endif
 
 #if defined(CONFIG_WEDGE) || defined(CONFIG_WEDGE100) || \
-	defined(CONFIG_CMM) || defined(CONFIG_FBTP) || defined(CONFIG_PWNEPTUNE) || \
+	defined(CONFIG_CMM) || defined(CONFIG_VOYAGER) || defined(CONFIG_FBTP) || defined(CONFIG_PWNEPTUNE) || \
 	defined(CONFIG_FBTTN) || defined(CONFIG_FBY2) || \
 	defined(CONFIG_YOSEMITE) || defined(CONFIG_MINIPACK) || \
 	defined(CONFIG_MINILAKETB) || defined(CONFIG_YAMP) || \
@@ -157,7 +157,7 @@ void __init ast_add_device_gmac(void)
 	// the latest open source drivers do not, so we drop the first
 	// MAC specs.
 #if !defined(CONFIG_WEDGE) && !defined(CONFIG_WEDGE100) && \
-    !defined(CONFIG_CMM) && !defined(CONFIG_MINIPACK) && \
+    !defined(CONFIG_CMM) && !defined(CONFIG_VOYAGER) && !defined(CONFIG_MINIPACK) && \
     !defined(CONFIG_GALAXY100)
 
 	ast_eth0_data.DF_support = !isRevA0;
diff --git a/drivers/net/ethernet/faraday/ftgmac100.c b/drivers/net/ethernet/faraday/ftgmac100.c
index d973bf0..23f6630 100644
--- a/drivers/net/ethernet/faraday/ftgmac100.c
+++ b/drivers/net/ethernet/faraday/ftgmac100.c
@@ -3261,7 +3261,7 @@ static int ftgmac100_open(struct net_device *netdev)
 
 #if defined(CONFIG_WEDGE) || defined(CONFIG_WEDGE100) || \
     defined(CONFIG_CMM) || defined (CONFIG_PWNEPTUNE) || \
-    defined(CONFIG_MINIPACK) || defined (CONFIG_GALAXY100) || \
+    defined(CONFIG_MINIPACK) || defined(CONFIG_VOYAGER) || defined (CONFIG_GALAXY100) || \
     defined(CONFIG_MINILAKETB)
 	ftgmac100_start_hw(priv, 1000);
 #elif defined(CONFIG_FBTP)
diff --git a/drivers/net/phy/broadcom.c b/drivers/net/phy/broadcom.c
index 6698743..33f694b 100644
--- a/drivers/net/phy/broadcom.c
+++ b/drivers/net/phy/broadcom.c
@@ -371,7 +371,7 @@ static int bcm54616s_config_aneg(struct phy_device *phydev)
 	ret = genphy_config_aneg(phydev);
 
 #if defined(CONFIG_WEDGE) || defined(CONFIG_WEDGE100) || defined(CONFIG_CMM) || \
-	defined(CONFIG_MINIPACK) || defined(CONFIG_YAMP) || \
+	defined(CONFIG_MINIPACK) || defined(CONFIG_VOYAGER) || defined(CONFIG_YAMP) || \
 	defined(CONFIG_GALAXY100)
 	/* disable GTXCLK Clock Delay */
 	phy_write(phydev, 0x1C, 0x8C00);
